src/QMCWaveFunctions/tests/test_einset_spinor.cpp

```diff
@@ -435,7 +435,86 @@ TEST_CASE("Einspline SpinorSet from HDF", "[wavefunction]")
 
     elec_.rejectMove(iat);
   }
+
+
+  // test batched interface
+  // first move elec_ back to original positions for reference
+  Rnew    = elec_.R - dR;
+  elec_.R = Rnew;
+  elec_.update();
+
+  //now create second walker, with permuted particle positions
+  ParticleSet elec_2(elec_);
+  // permute electrons
+  elec_2.R[0]     = elec_.R[1];
+  elec_2.R[1]     = elec_.R[2];
+  elec_2.R[2]     = elec_.R[0];
+  elec_2.spins[0] = elec_.spins[1];
+  elec_2.spins[1] = elec_.spins[2];
+  elec_2.spins[2] = elec_.spins[0];
+  RefVectorWithLeader<ParticleSet> p_list(elec_);
+  p_list.push_back(elec_);
+  p_list.push_back(elec_2);
+
+  std::unique_ptr<SPOSet> spo_2(spo->makeClone());
+  RefVectorWithLeader<SPOSet> spo_list(*spo);
+  spo_list.push_back(*spo);
+  spo_list.push_back(*spo_2);
+
```

Comment:
I will be better to initialize shared resource for particleset to avoid potential issues with GPUs
https://github.com/QMCPACK/qmcpack/blob/908bcb018cfab78bf15295a881fb4e8f1c83d383/src/QMCWaveFunctions/tests/test_TrialWaveFunction.cpp#L208

Reply:
Done. also added as well as the ResourceCollectionTeamLock which I saw was also used in test_TrialWaveFunction